packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -142,21 +143,12 @@ export const SelectionHandler = (
       updated: new Date()
     };
 
+    /**
+     * During mouse selection on the desktop, annotation won't usually exist while the selection is being edited.
+     * But it will be typical during keyboard or mobile handlebars selection!
+     */
     if (store.getAnnotation(currentTarget.annotation)) {
       store.updateTarget(currentTarget, Origin.LOCAL);
-    } else {
-      // Proper lifecycle management: clear selection first...
-      selection.clear();
```

Comment:
I think that we may need to restore the `selection.clear()` call if there's nothing to update. Because now the previous, discarded selection stays up until you create a brand-new one.

https://github.com/user-attachments/assets/cb0d6edc-5a48-400b-90f8-a61c41251200

Reply:
UPD: Submitted a PR - https://github.com/recogito/text-annotator-js/pull/155